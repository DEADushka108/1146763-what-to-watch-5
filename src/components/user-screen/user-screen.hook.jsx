import React, {useEffect} from 'react';
import MoviesList from '../movies-list/movies-list';
import PropTypes from 'prop-types';
import {movieDetails} from '../../types/types';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../utils/const';
import {getFavoriteList} from '../../store/user/selectors';
import {connect} from 'react-redux';
import UserBlock from '../user-block/user-block';
import {Operation as UserOperation} from '../../store/user/user.js';

const UserScreen = (props) => {
  const {moviesList, loadFavoriteList} = props;

  useEffect(() => {
    loadFavoriteList();
  }, [moviesList.length]);

  return <React.Fragment>
    <div className="user-page">
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock/>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <MoviesList movies={moviesList} count={moviesList.length}/>
      </section>

      <footer className="page-footer">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link logo__link--light">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="copyright">
          <p>Â© 2020 What to watch Ltd.</p>
        </div>
      </footer>
    </div>
  </React.Fragment>;
};

UserScreen.propTypes = {
  moviesList: PropTypes.arrayOf(movieDetails).isRequired,
  loadFavoriteList: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  moviesList: getFavoriteList(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadFavoriteList() {
    dispatch(UserOperation.loadFavoriteList());
  }
});

export {UserScreen};
export default connect(mapStateToProps, mapDispatchToProps)(UserScreen);

