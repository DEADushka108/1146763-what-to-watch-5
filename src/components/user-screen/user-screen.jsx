import React from 'react';
import MoviesList from '../movies-list/movies-list';
import PropTypes from 'prop-types';
import {movieDetails} from '../../types/types';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../utils/const';

const UserScreen = (props) => {
  const {moviesList, onMovieCardClick} = props;
  return <React.Fragment>
    <div className="user-page">
      <header className="page-header user-page__head">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <h1 className="page-title user-page__title">My list</h1>

        <div className="user-block">
          <div className="user-block__avatar">
            <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
          </div>
        </div>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <MoviesList movies={moviesList} count={moviesList.length} onClick={onMovieCardClick}/>
      </section>

      <footer className="page-footer">
        <div className="logo">
          <Link to={AppRoute.ROOT} className="logo__link logo__link--light">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="copyright">
          <p>Â© 2019 What to watch Ltd.</p>
        </div>
      </footer>
    </div>
  </React.Fragment>;
};

UserScreen.propTypes = {
  moviesList: PropTypes.arrayOf(movieDetails).isRequired,
  onMovieCardClick: PropTypes.func.isRequired,
};

export default UserScreen;
