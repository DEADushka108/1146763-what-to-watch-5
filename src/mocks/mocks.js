export * from './const/const.js';
export * from './movies/movies.js';
export * from './reviews/reviews.js';
export * from './store/store.js';
export * from './user/user.js';
