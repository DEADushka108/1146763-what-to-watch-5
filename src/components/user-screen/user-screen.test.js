import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import {moviesList, store, noop} from '../../mocks/mocks.js';
import {UserScreen} from './user-screen.jsx';

describe(`Render UserScreen`, () => {
  it(`Should render UserScreen component correctly`, () => {
    const tree = renderer
      .create((
        <Provider store={store}>
          <MemoryRouter>
            <UserScreen
              moviesList={moviesList}
              loadFavoriteList={noop}
            />
          </MemoryRouter>
        </Provider>
      ), {
        createNodeMock() {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
