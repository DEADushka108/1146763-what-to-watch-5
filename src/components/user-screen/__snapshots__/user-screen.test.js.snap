// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserScreen Should render UserScreen component correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="batman"
            height="63"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/4.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/macbeth.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/we-need-to-talk-about-kevin.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            We need to talk about Kevin
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/shutter-island.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/4"
            onClick={[Function]}
          >
            Shutter Island
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/snatch.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/5"
            onClick={[Function]}
          >
            Snatch
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/pulp-fiction.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/6"
            onClick={[Function]}
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/revenant.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/7"
            onClick={[Function]}
          >
            Revenant
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            muted={true}
            poster="img/bohemian-rhapsody.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/8"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2020 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
